examples/cli.rs

```diff
@@ -237,25 +248,75 @@ async fn receive<C: Store + MessageStore>(
                         "Quote from {:?}: > {:?} / {}",
                         metadata.sender,
                         quote,
-                        message.body(),
+                        message.body()
                     );
                 } else if let Some(reaction) = message.reaction {
                     println!(
                         "Reaction to message sent at {:?}: {:?}",
-                        reaction.target_sent_timestamp, reaction.emoji,
-                    )
+                        reaction.target_sent_timestamp, reaction.emoji
+                    );
                 } else {
-                    println!("Message from {:?}: {:?}", metadata, message);
-                    // fetch the groups v2 info here, just for testing purposes
+                    let self_is_sender = metadata.sender.uuid == Some(manager.uuid());
+                    if self_is_sender {
+                        println!("Message sent to {:?}: {:?}", metadata.sender, message);
+                    } else {
+                        println!("Message from {:?}: {:?}", metadata.sender, message);
+                    }
+                    let body = message.body().to_string();
+                    let mut session_name: String = "Unknown contact".to_string();
+                    // fetch the groups v2 info here
                     if let Some(group_v2) = message.group_v2 {
                         let master_key_bytes: [u8; 32] =
                             group_v2.master_key.clone().unwrap().try_into().unwrap();
                         let master_key = GroupMasterKey::new(master_key_bytes);
                         let group = manager.get_group_v2(master_key).await?;
                         let group_changes = manager.decrypt_group_context(group_v2)?;
-                        println!("Group v2: {:?}", group.title);
+                        session_name = group.title;
+                        println!("Group v2: {:?}", session_name);
                         println!("Group change: {:?}", group_changes);
                         println!("Group master key: {:?}", hex::encode(&master_key_bytes));
+                    } else if !self_is_sender {
+                        //get the contact name
+                        match metadata.sender.uuid {
+                            Some(uuid) => {
+                                println!("uuid: {:?}", uuid);
+                                for contact in manager
```

Comment:
I feel like we should find a way to put this inside of `Manager` since IMO this is what `presage` is about. I'm fine merging this as-is for now, though. I'll try to see if there's a way to include it inside.

Reply:
I agree, but it should be optional, because maybe it's used on a headless server without the need for notifications.